components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -338,4 +344,33 @@ public static boolean isAuthHeaderMatch(MessageContext messageContext, String au
         }
         return false;
     }
+
+    /**
+     * Build configurations of endpoints which are allowed to skip authorization with particular auth handler.
+     */
+    public void buildSkipAuthorizationAllowedEndpointsData() {
+
+        OMElement skipAuthorizationConfig = IdentityConfigParser.getInstance().getConfigElement(
+                AUTHORIZATION_CONTROL_ELE);
+        if (skipAuthorizationConfig != null) {
+            Iterator<OMElement> configs = skipAuthorizationConfig.getChildrenWithName(
+                    new QName(SKIP_AUTHORIZATION_ELE));
+            if (configs != null) {
+                while (configs.hasNext()) {
+                    OMElement config = configs.next();
+                    String authHandlerName = config.getAttributeValue(new QName(AUTH_HANDLER_ELE));
+                    String[] allowedEndpoints = config.getAttributeValue(
+                            new QName(ENDPOINT_LIST_ELE)).split(regex);
+                    skipAuthorizationAllowedEndpoints.put(authHandlerName, allowedEndpoints);
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieve Map of endpoint that allowed to skip authorization against the auth handler.
+     */
+    public Map<String, String[]> getSkipAuthorizationAllowedEndpoints() {
```

Comment:
Generate the method comment properly. please check other places as well.

Reply:
Addressed